apps/web/src/components/Onboarding/VerificationFlow/VerifySeedphraseModal.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  Button,
+  Center,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  Heading,
+  Icon,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  Text,
+} from "@chakra-ui/react";
+import { MnemonicAutocomplete, useDynamicModalContext } from "@umami/components";
+import { selectRandomElements } from "@umami/core";
+import { accountsActions, useAppDispatch, useCurrentAccount } from "@umami/state";
+import { useState } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+
+import { KeyIcon } from "../../../assets/icons";
+import { IS_DEV } from "../../../env";
+import { useColor } from "../../../styles/useColor";
+import { ModalBackButton } from "../../BackButton";
+import { ModalCloseButton } from "../../CloseButton";
+
+type VerifySeedphraseModalProps = {
+  seedPhrase: string;
+};
+
+export const VerifySeedphraseModal = ({ seedPhrase }: VerifySeedphraseModalProps) => {
+  const color = useColor();
+  const dispatch = useAppDispatch();
+  const currentAccount = useCurrentAccount()!;
+  const { onClose } = useDynamicModalContext();
+  const form = useForm({
+    mode: "onBlur",
+  });
+  const {
+    handleSubmit,
+    formState: { errors, isValid },
+  } = form;
+
+  const seedphraseArray = seedPhrase.split(" ");
+  const [randomElements] = useState(selectRandomElements(seedphraseArray, 3));
+
+  const onSubmit = () => {
+    dispatch(
+      accountsActions.setIsVerified({
+        pkh: currentAccount.address.pkh,
+        isVerified: true,
+      })
+    );
+
+    onClose();
```

Comment:
we must also clean the password in the accountsSlice, otherwise we have a plain text password hanging in memory until the page refresh